From e86520793fac1cc9cb4b779b702a519d378f1d33 Mon Sep 17 00:00:00 2001
From: John Brooks <john.brooks@jollamobile.com>
Date: Wed, 6 Nov 2013 02:14:53 -0700
Subject: [PATCH 4/7] Continue reading from plugins after handling response if
 necessary

There may be more than one response from the plugin; for example, during
a refresh, you receive a store and result response. If both arrive
simultaniously, readyRead() will not be signalled again after handling
the first response. The request will then be deadlocked waiting for a
response that is sitting in the file buffer.

The immediate solution is simply to read again if there's more to be
read after handling a response.
---
 src/signond/pluginproxy.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/signond/pluginproxy.cpp b/src/signond/pluginproxy.cpp
index b088723..4c207a3 100644
--- a/src/signond/pluginproxy.cpp
+++ b/src/signond/pluginproxy.cpp
@@ -436,6 +436,10 @@ void PluginProxy::handlePluginResponse(const quint32 resultOperation,
     }
 
     connect(m_process, SIGNAL(readyRead()), this, SLOT(onReadStandardOutput()));
+    if (m_process->bytesAvailable()) {
+        TRACE() << "plugin has more to read after handling a response";
+        onReadStandardOutput();
+    }
 }
 
 void PluginProxy::onReadStandardError()
-- 
2.20.1

